tests/test_models.py

```diff
@@ -167,6 +167,15 @@ def test_custom_render_variations(self, db):
         assert instance.image.thumbnail.width == 100
         assert instance.image.thumbnail.height == 100
 
+    def test_defer(self, db, django_assert_num_queries):
+        """Test that set_variations doesn't attempt to use field data
```

Comment:
The docstring is not pep257 compliant, maybe:
```
"""
`set_variations` does not access a deferred field.

Accessing a deferred field would cause Django to do
a second implicit database query.
```